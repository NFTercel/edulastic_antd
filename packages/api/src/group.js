import API from './utils/API';

const api = new API();
const prefix = '/group';

const fetchMyGroups = () =>
  api
    .callApi({
      url: `${prefix}/mygroups`,
      method: 'get'
    })
    .then(result => result.data.result);

export default {
  fetchMyGroups
};
