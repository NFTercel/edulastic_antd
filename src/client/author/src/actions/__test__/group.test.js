import { fetchGroupsAction, fetchStudentsOfGroupAction } from "../group";

describe("group", () => {
  it("fetch groups should return an action", () => {
    expect(fetchGroupsAction()).toMatchSnapshot();
  });

  it("fetch students of group action should return an action", () => {
    expect(
      fetchStudentsOfGroupAction({
        id: "53wqwri21567"
      })
    ).toMatchSnapshot();
  });
});
