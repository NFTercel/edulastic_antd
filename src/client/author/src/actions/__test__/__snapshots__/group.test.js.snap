// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group fetch groups should return an action 1`] = `
Object {
  "type": "[group] fetch owner groups",
}
`;

exports[`group fetch students of group action should return an action 1`] = `
Object {
  "payload": Object {
    "id": "53wqwri21567",
  },
  "type": "[group] fetch students of group",
}
`;
